### ISSUE TITLE
Bug: [no-unused-vars] `reportUsedIgnorePattern` does not work for JSX Identifiers

### ISSUE THREAD

**Issue by bradzacher** [eslint/eslint#20942]
### Environment

ESLint Playground

### What parser are you using?

Default (Espree)

### Playground Code
```js
/*eslint no-unused-vars: ["error", { "reportUsedIgnorePattern": true, "varsIgnorePattern": "^_" }]*/

const x = { _Foo: 1, _Bar: () => null };
const { _Foo } = x;
export const y = _Foo;

<_Bar />
 
```

### ESLint Configuration 
<details> 
 <summary> Tap to see full config </summary> 

```js
import { defineConfig } from "eslint/config";

export default defineConfig([
    {
        "languageOptions": {
            "parserOptions": {
                "ecmaFeatures": {
                    "jsx": true
                }
            }
        }
    }
]);
```
 </details>

### ESLint Output
```
4:9 '_Foo' is marked as ignored but is used. Used vars must not match /^_/u. (no-unused-vars)
```

### What did you expect to happen?

`_Bar` should be reported just like `_Foo`

### What actually happened?

`_Bar` was not reported.

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tdW51c2VkLXZhcnM6IFtcImVycm9yXCIsIHsgXCJyZXBvcnRVc2VkSWdub3JlUGF0dGVyblwiOiB0cnVlLCBcInZhcnNJZ25vcmVQYXR0ZXJuXCI6IFwiXl9cIiB9XSovXG5cbmNvbnN0IHggPSB7IF9Gb286IDEsIF9CYXI6ICgpID0+IG51bGwgfTtcbmNvbnN0IHsgX0ZvbyB9ID0geDtcbmV4cG9ydCBjb25zdCB5ID0gX0ZvbztcblxuPF9CYXIgLz5cbiAiLCJvcHRpb25zIjp7InJ1bGVzIjp7fSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7ImpzeCI6dHJ1ZX19fX19

### Participation

- [x] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Comment by bradzacher** [comment]
Sorry I realised I incorrectly built the playground out when confirming the bug.
It doesn't repro in the playground, only in our older version of the rule.
